### DISCUSSION TITLE
Sunbird-Lern Create Org API

### DISCUSSION THREAD

**Discussion by devaj04** [Sunbird-Lern/Community#107] [Query] [answered]
Following the below URL and trying to create a new Org in the local system.
https://lern.sunbird.org/use/developer-installation/user-and-org-service/installation-guide

URL - localhost:9000/v1/org/create
Request Body
{
  "request":
   {
        "orgName": "localrootorg",
        "channel": "ChannelLocal",
        "description": "Description",
        "externalId": "localrootorg",
        "email": "info@org.org",
        "isSSOEnabled": true,
        "organisationType": "school",
        "isTenant": true,
        "id":"4551b384-3e96-480e-a9dd-75149990e678",
        "rootOrgId":"01",
        "userId":"devaj",
        "requestedBy":"Dev"
    }
}

Response Body
{
    "id": "api.org.create",
    "ver": "v1",
    "ts": "2023-04-20 09:45:04:034+0530",
    "params": {
        "resmsgid": "e88be4df-fc61-4822-a8a0-5d9533ecb325",
        "msgid": "e88be4df-fc61-4822-a8a0-5d9533ecb325",
        "err": "UOS_ORGCRT0012",
        "status": "FAILED",
        "errmsg": "Channel Registration failed."
    },
    "responseCode": "SERVER_ERROR",
    "result": {}
}

what value we have to set in **EKSTEP_BASE_URL** , unable to find in documents
<img width="629" alt="image" src="https://user-images.githubusercontent.com/125238862/233259134-0842c021-f7b8-41d0-a455-73abf9d9f104.png">

**Comment by reshmi-nair** [comment]
@devaj04 
You will have to configure the learning-service or content-service url which has channel APIs, for updating or creating channel data while creating and org.
eg : ekstep_api_base_url=http://11.3.3.22:8080/learning-service

ekstep.channel.reg.api.url=/channel/v3/create
ekstep.channel.update.api.url=/channel/v3/update

**Reply by devaj04** [reply]
@reshmi-nair 
All the application is running on the same port 9000
so could you please suggest how can we run 2 or more services on the same machine?

**Reply by Jayaprakash8887** [reply, marked as answer]
@devaj04  ports can be changed in 'application.conf' file of respective services.

```
play.filters {
  hosts {
    # Allow requests to example.com, its subdomains, and localhost:9000.
    allowed = ["localhost:9000","."]
  }
```

**Reply by Jayaprakash8887** [reply]
@devaj04 For local setup, instead of attempting to connecting to channel service, pls hard code expected response and continue. **(Kindly remember to undo it while raising a PR)**

**Reply by ashima-del** [reply]
even after changing port in application.conf I am getting error:
play.filters {
  hosts {
    # Allow requests to example.com, its subdomains, and localhost:9000.
    allowed = ["localhost:8080","."]
  }


[error] p.c.s.NettyServer - Failed to listen for HTTP on /0.0.0.0:9000!
